src/litserve/server.py

```diff
@@ -390,17 +390,31 @@ async def stream_predict(request: self.request_type) -> self.response_type:
                 self.app.add_middleware(middleware)
 
     @staticmethod
-    def generate_client_file():
+    def generate_client_file(port: Union[str, int] = 8000):
         src_path = os.path.join(os.path.dirname(__file__), "python_client.py")
         dest_path = os.path.join(os.getcwd(), "client.py")
 
         if os.path.exists(dest_path):
             return
 
-        # Copy the file to the destination directory
         try:
+            # Copy the file to the destination directory
             shutil.copy(src_path, dest_path)
             print(f"File '{src_path}' copied to '{dest_path}'")
+
+            # After copying, close the client file and replace the default port
+            with open(dest_path) as f:
+                client_code = f.read()
+
+            # Replace the placeholder port with the actual port specified by user via Server.run()
+            client_code = client_code.replace("{{PORT}}", str(port))
+
+            # Write the updated code back to the client file
+            with open(dest_path, "w") as f:
+                f.write(client_code)
+
+            print(f"Port {port} has been set in 'dest_path'")
+
```

Comment:
we can read the `src_path`, update the {PORT} and write to the `dest_path`. 

It will reduce the number of disk writes from 2 to 1.

Reply:
good point! done